packages/vite/src/node/plugins/css.ts

```diff
@@ -885,38 +926,73 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
         })
       }
 
-      function extractCss() {
+      async function extractCss() {
         let css = ''
-        const collected = new Set<OutputChunk>()
+        let line = 0
+        const collected = new Set<OutputAsset | OutputChunk>()
+        const concatCssEndLines: Array<{ file: string; end: number }> = []
         const prelimaryNameToChunkMap = new Map(
           Object.values(bundle)
             .filter((chunk): chunk is OutputChunk => chunk.type === 'chunk')
             .map((chunk) => [chunk.preliminaryFileName, chunk]),
         )
 
-        function collect(fileName: string) {
+        function collect(fileName: string): string {
           const chunk = bundle[fileName]
-          if (!chunk || chunk.type !== 'chunk' || collected.has(chunk)) return
-          collected.add(chunk)
+          if (!chunk || chunk.type !== 'chunk') {
+            collected.add(chunk)
+            return ''
+          }
+          if (collected.has(chunk)) {
+            return ''
+          }
+
+          const css = chunkCSSMap.get(chunk.preliminaryFileName) ?? ''
+          const importedCSS = chunk.imports.reduce((css, file) => {
+            const imported = collect(file)
+            if (imported.length > 0) {
+              return css + '\n' + imported
```

Comment:
I think the CSS files that is concatenated here should be included in `concatCssEndLines`, otherwise the position won't be mapped to those files.

Reply:
Yes, I agree with you. I missed that part.